src/test/java/seedu/rc4hdb/model/venues/VenueTest.java

```diff
@@ -0,0 +1,248 @@
+    // after adding it to testVenue in a particular test case.
+
+    // However, to avoid the possibility of breaking the entire set of tests with just one errand test case,
+    // we have decided to re-instantiate a new Venue each time.
+
+    // Some checks are enforced to avoid re-instantiating a new Venue instance for a correct testVenue
+
+    @BeforeEach
+    public void initEach() {
+        if (isDefaultTestVenue(testVenue)) {
+            return;
+        }
+        testVenue = new VenueBuilder(MEETING_ROOM)
+                .withBookings(MR_ALICE_MONDAY_5_TO_6PM)
+                .build();
+    }
+
+    private boolean isDefaultTestVenue(Venue someVenue) {
+        if (someVenue.getVenueName() != MEETING_ROOM_VENUE_NAME) {
+            return false;
+        }
+        if (!someVenue.getObservableBookings().equals(List.of(MR_ALICE_MONDAY_5_TO_6PM))) {
+            return false;
+        }
+        return true;
+    }
+
+    private BookingDescriptor getBookingDescriptorOf(Booking booking) {
+        BookingDescriptor descriptor = new BookingDescriptor();
+        descriptor.setVenueName(booking.getVenueName());
+        descriptor.setResident(booking.getResident());
+        descriptor.setHourPeriod(booking.getHourPeriod());
+        descriptor.setDayOfWeek(booking.getDayOfWeek());
+        return descriptor;
+    }
+
+    // TODO: Should add requireNonNull checks for Venue class
+    // TODO: Update the equals method for the Venue class
+
+    // Test the first constructor
+    @Test
+    public void constructor_withoutListOfBookings_returnsVenueWithNoBookings() {
+        Venue venue = new Venue(MEETING_ROOM_VENUE_NAME);
+        assertEquals(venue.getObservableBookings(), Collections.emptyList());
+    }
+
+    // Test the second constructor
+    @Test
+    public void constructor_withListOfBookings_returnsVenueWithSameListOfBookings() {
+        List<Booking> bookings = Arrays.asList(MR_ALICE_MONDAY_5_TO_6PM, MR_BOB_TUESDAY_6_TO_7PM);
+        Venue venue = new Venue(bookings, MEETING_ROOM_VENUE_NAME);
+        assertEquals(venue.getObservableBookings(), bookings);
+    }
+
+    // Null input for first constructor throws NullPointerException
+    @Test
+    public void constructor_withNullVenueName_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Venue(null));
+    }
+
+    // At least one null input for second constructor throws NullPointerException
+    @Test
+    public void constructor_withNullVenueNameOrNullBookings_throwsNullPointerException() {
+        List<Booking> bookings = Arrays.asList(MR_ALICE_MONDAY_5_TO_6PM, MR_BOB_TUESDAY_6_TO_7PM);
+        assertThrows(NullPointerException.class, () -> new Venue(bookings, null));
+        assertThrows(NullPointerException.class, () -> new Venue(null, MEETING_ROOM_VENUE_NAME));
+        assertThrows(NullPointerException.class, () -> new Venue(null, null));
```

Comment:
I did not know that running multiple assertions in a method is possible. After some searching, I am leaning towards not practicing this, as if the test fails, it would be harder to determine the cause of the failure.

Though some people do support this practice. You can read more about this issue [here](https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test)

Reply:
Thank you for doing research on this. 

I respectfully disagree that the cause of failure would be hard to determine, given that we are ultimately only testing the Venue constructor in this method.

The intention of this test case is to check for the ability of the Venue constructor to reject any and all null inputs. Bundling these `assertThrows` statements within the same method was also intentional, as the method should fail if even one null input was not properly validated. 

From a readability perspective, it could also be argued that this method alone is sufficient to inform a developer that the null checks are robust. The same could not be said if each of these `assertThrows` statements were isolated within their own test case.

All said, I do agree that there are cases where using multiple assertion statements within a test case could be counter-productive. 😄